projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php

```diff
@@ -191,13 +191,21 @@ function_exists( 'wpcom_site_has_feature' ) &&
 	 * Adds the site switcher link if user has more than one site.
 	 */
 	public function add_browse_sites_link() {
-		$site_count = get_user_option( 'wpcom_site_count' );
-		if ( ! $site_count || $site_count < 2 ) {
+		if ( 'wp-admin' === get_option( 'wpcom_admin_interface' ) ) {
+			add_menu_page(
+				esc_attr__( 'All Sites', 'jetpack' ),
+				esc_attr__( 'All Sites', 'jetpack' ),
+				'manage_options',
+				'https://wordpress.com/sites',
+				null,
+				'dashicons-arrow-left-alt2',
+				0
+			);
 			return;
 		}
 
-		// Unnecessary because "My Sites" always links to the Sites page.
-		if ( 'wp-admin' === get_option( 'wpcom_admin_interface' ) ) {
```

Comment:
I think this is really the ONLY change we need here. Remove the early return based on this option. That would mean that:
* Sites without nav redesign on 'classic' will get the browse sites link back.
* Sites on nav redesign wont be effected since this module isn't loaded.

Reply:
True, I updated the PR with your suggestions